File: lapi/internal/modules/prestation/types/prestation_types.go
package types

import (
	"time"

	"github.com/google/uuid"
)

type NoBody struct{}

type SlotIDPath struct {
	SlotID string `path:"slotId"`
}

type BookingIDPath struct {
	BookingID string `path:"bookingId"`
}

// ── Responses ───────────────────────────────────────────────────────

type SlotResponse struct {
	ID          uuid.UUID `json:"id"`
	Date        time.Time `json:"date"`
	TimeSlot    string    `json:"time_slot"`
	IsAvailable bool      `json:"is_available"`
}

type BookingResponse struct {
	ID                uuid.UUID     `json:"id"`
	UserID            uuid.UUID     `json:"user_id"`
	OrderID           uuid.UUID     `json:"order_id"`
	Slot              *SlotResponse `json:"slot,omitempty"`
	AddressStreet     string        `json:"address_street"`
	AddressCity       string        `json:"address_city"`
	AddressPostalCode string        `json:"address_postal_code"`
	GuestCount        int           `json:"guest_count"`
	Notes             string        `json:"notes"`
	ChefNotes         string        `json:"chef_notes,omitempty"`
	Status            string        `json:"status"`
	CreatedAt         time.Time     `json:"created_at"`
}

// ── Admin requests ──────────────────────────────────────────────────

type CreateSlotRequest struct {
	Date     string `json:"date" validate:"required"`      // YYYY-MM-DD
	TimeSlot string `json:"time_slot" validate:"required"` // HH:MM-HH:MM (free format)
}

type BlockSlotInput struct {
	SlotIDPath
}

type UnblockSlotInput struct {
	SlotIDPath
}

// ── Admin booking management ────────────────────────────────────────

type UpdateBookingStatusInput struct {
	BookingIDPath
	Status string `json:"status" validate:"required,oneof=confirmed completed cancelled"`
}

type UpdateChefNotesInput struct {
	BookingIDPath
	Notes string `json:"notes"`
}

type CancelBookingInput struct {
	BookingIDPath
	Reason string `json:"reason"`
	Refund bool   `json:"refund"`
}

type NotifyBookingInput struct {
	BookingIDPath
}

// ── Extended admin booking response ────────────────────────────────

type BookingAdminResponse struct {
	BookingResponse
	ChefNotes string `json:"chef_notes,omitempty"`
}

File: lapiweb/types/prestationTypes.ts
export interface SlotResponse {
  id: string;
  date: string;
  time_slot: string;
  is_available: boolean;
}

export interface BookingResponse {
  id: string;
  user_id: string;
  order_id: string;
  slot: SlotResponse | null;
  address_street: string;
  address_city: string;
  address_postal_code: string;
  guest_count: number;
  notes: string;
  status: "confirmed" | "completed" | "cancelled";
  created_at: string;
}

export interface PrestationPricing {
  base_price: number;
  price_per_person: number;
  min_guests: number;
  max_guests: number;
}

export interface CreateSlotRequest {
  date: string;
  time_slot: string; // HH:MM-HH:MM
}
